synapse/util/__init__.py

```diff
@@ -112,17 +114,30 @@ def batch_iter(iterable, size):
     return iter(lambda: tuple(islice(sourceiter, size)), ())
 
 
-def _log_exception_wrapper(f):
-    """Used to wrap looping calls to log loudly if they get killed
+def make_log_failure_errback(msg, consumeErrors=True):
```

Comment:
could you not have `msg` and `consumeErrors` as params to `log_failure` and then just do `addErrback(log_failure, msg)` ?

Reply:
... yes.

This PR is going terribly.